### PR TITLE
T21345/reingest

### PR DESCRIPTION
**Pull request by manuq**
https://phabricator.endlessm.com/T21345

### CODE REVIEW

**Comment by sgnn7** [comment]
LGTM but I'll leave the full CR and merge to @dnohales

File: lib/index.js [hunk 1]
```diff
@@ -1097,23 +1131,19 @@ class Hatch {
...
-            const no_tgz = argv.indexOf('--no-tgz');
-            if (no_tgz >= 0) {
-                options.no_tgz = true;
-            }
+        const argv_options = options.argv || process.argv.slice(2);
+
+        const config_options = argv(ARGV_CONFIGFILE_OPTIONS, argv_options);
+        if (config_options['config-file']) {
+            const cf = config_options['config-file'];
+            const file_options = JSON.parse(fs.readFileSync(cf, 'utf8'));
```
Comment (dnohales) [inline lib/index.js]: The package `fs-extra` has a function `readJsonSync`.

File: lib/index.js [hunk 2]
```diff
@@ -1097,23 +1131,19 @@ class Hatch {
...
+            options = Object.assign(options, file_options);
         }
 
-        return options;
+        return Object.assign(options, argv(ARGV_OPTIONS, argv_options))
```
Comment (dnohales) [inline lib/index.js]: You could easily have only one `ARGV_OPTIONS` variable with `config-file` in it, use it in line 1139, and here do:

```
return Object.assign(file_options, config_options, options);
```

(arguments on the right have priority)

File: lib/index.js [hunk 3]
```diff
@@ -1097,23 +1131,19 @@ class Hatch {
...
+    _parse_options(options) {
+        options = options || {};
 
-            const no_tgz = argv.indexOf('--no-tgz');
-            if (no_tgz >= 0) {
-                options.no_tgz = true;
-            }
+        const argv_options = options.argv || process.argv.slice(2);
+
+        const config_options = argv(ARGV_CONFIGFILE_OPTIONS, argv_options);
```
Comment (dnohales) [inline lib/index.js]: I'd name this `cli_options`

File: lib/index.js [hunk 4]
```diff
@@ -6,12 +6,31 @@ const crypto = require('crypto');
...
+};
+
+const ARGV_OPTIONS = {
+    'no-tgz': {
+        type: 'boolean'
+    },
+    'path': {
+        type: 'string'
+    },
+    'urls': {
```
Comment (dnohales) [inline lib/index.js]: Maybe `reingest-urls`? Not a blocker though.

Comment (manuq) [reply lib/index.js]: "reingest" is a good verb for SOMA, but the purpose of 'urls' can be any, like ingesting a specific URL for debugging.

**Pull request merged**